modules/src/mock/context.rs

```diff
@@ -563,27 +582,27 @@ pub struct MockIbcStore {
     pub channel_ids_counter: u64,
 
     /// All the channels in the store. TODO Make new key PortId X ChanneId
-    pub channels: BTreeMap<(PortId, ChannelId), ChannelEnd>,
+    pub channels: PortChannelIdMap<ChannelEnd>,
 
     /// Tracks the sequence number for the next packet to be sent.
-    pub next_sequence_send: BTreeMap<(PortId, ChannelId), Sequence>,
+    pub next_sequence_send: PortChannelIdMap<Sequence>,
 
     /// Tracks the sequence number for the next packet to be received.
-    pub next_sequence_recv: BTreeMap<(PortId, ChannelId), Sequence>,
+    pub next_sequence_recv: PortChannelIdMap<Sequence>,
 
     /// Tracks the sequence number for the next packet to be acknowledged.
-    pub next_sequence_ack: BTreeMap<(PortId, ChannelId), Sequence>,
+    pub next_sequence_ack: PortChannelIdMap<Sequence>,
 
-    pub packet_acknowledgement: BTreeMap<(PortId, ChannelId, Sequence), AcknowledgementCommitment>,
+    pub packet_acknowledgement: PortChannelIdMap<BTreeMap<Sequence, AcknowledgementCommitment>>,
 
     /// Maps ports to the the module that owns it
     pub port_to_module: BTreeMap<PortId, ModuleId>,
 
     /// Constant-size commitments to packets data fields
-    pub packet_commitment: BTreeMap<(PortId, ChannelId, Sequence), PacketCommitment>,
+    pub packet_commitment: PortChannelIdMap<BTreeMap<Sequence, PacketCommitment>>,
 
     // Used by unordered channel
-    pub packet_receipt: BTreeMap<(PortId, ChannelId, Sequence), Receipt>,
+    pub packet_receipt: PortChannelIdMap<BTreeMap<Sequence, Receipt>>,
```

Comment:
So far `PortChannelIdMap` is just a type alias, so I was hesitant to add method API around it. Maybe standalone helper fns would be the simplest way to deal with it; I don't want to invent an extension trait or do a newtype just for a convenience tradeoff.

Reply:
> I think it would be better to use `BTreeMap<ibc::core::ics24_host::path::ReceiptsPath, Receipt>`

A non-allocating API for this would force use of `ReceiptsPath` in method parameters for `get_packet_receipt`/`store_packet_receipt`, and the call sites do not readily support that: the `PortId`, `ChannelId`, `Sequence` triplet comes from some larger structure like `Packet`. I suppose grouping of these packet fields as `ReceiptsPath` inside `Packet` does not make sense, because it might as well be an `AcksPath`?

Reply:
Yeah, it's tricky to design a non-allocating API, also because some paths deal with source port/channel-id (e.g. `CommitmentsPath`) and while others need destination port/channel-id (e.g. `ReceiptsPath` & `AcksPath`). Maybe we should use references in all path structs in the future ->
```rust
pub struct ReceiptsPath<'a> {
    pub port_id: &'a PortId,
    pub channel_id: &'a ChannelId,
    pub sequence: Sequence,
}
```

Reply:
I gave it some thought and tried to work with some reference-containing structs too, but quickly gave it up as impractical.

Oddly, it's one instance where the internal implementation, in conjunction with Rust associative container lookup APIs, comes at odds with API ergonomics. If anyone can suggest how to organize multi-value key containers better, let it be known.